oauth2_http/java/com/google/auth/oauth2/Slf4jLoggingHelpers.java

```diff
@@ -0,0 +1,180 @@
+import com.google.gson.Gson;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import org.slf4j.Logger;
+
+/** Contains helper methods to log auth requests and responses */
+class Slf4jLoggingHelpers {
+  private static final Gson gson = new Gson();
+  private static final Set<String> SENSITIVE_KEYS = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+
+  static {
+    SENSITIVE_KEYS.addAll(
+        Arrays.asList(
+            "token",
+            "assertion",
+            "access_token",
+            "client_secret",
+            "refresh_token",
+            "signedBlob",
+            "authorization"));
+  }
+
+  static void logRequest(HttpRequest request, LoggerProvider loggerProvider, String message) {
+    try {
+      Logger logger = loggerProvider.getLogger();
+      if (logger.isInfoEnabled()) {
+        Map<String, Object> loggingDataMap = new HashMap<>();
+        loggingDataMap.put("request.method", request.getRequestMethod());
+        loggingDataMap.put("request.url", request.getUrl().toString());
+
+        Map<String, Object> headers = new HashMap<>();
+        request
+            .getHeaders()
+            .forEach(
+                (key, val) -> {
+                  if (SENSITIVE_KEYS.contains(key)) {
+                    String hashedVal = calculateSHA256Hash(String.valueOf(val));
+                    headers.put(key, hashedVal);
+                  } else {
+                    headers.put(key, val);
+                  }
+                });
+        loggingDataMap.put("request.headers", gson.toJson(headers));
+
+        if (request.getContent() != null && logger.isDebugEnabled()) {
+          // are payload always GenericData? If so, can parse and store in json
+          if (request.getContent() instanceof UrlEncodedContent) {
+            // this is parsed to GenericData because that is how it is constructed.
+            GenericData data = (GenericData) ((UrlEncodedContent) request.getContent()).getData();
+            Map<String, Object> contextMap = parseGenericData(data);
+            loggingDataMap.put("request.payload", gson.toJson(contextMap));
+          } else if (request.getContent() instanceof JsonHttpContent) {
+            String jsonData = gson.toJson(((JsonHttpContent) request.getContent()).getData());
+            loggingDataMap.put("request.payload", jsonData);
+          }
+
+          Slf4jUtils.log(logger, org.slf4j.event.Level.DEBUG, loggingDataMap, message);
+        } else {
+
+          Slf4jUtils.log(logger, org.slf4j.event.Level.INFO, loggingDataMap, message);
+        }
+      }
+    } catch (Exception e) {
+      // let logging fail silently
+    }
+  }
+
+  static void logResponse(HttpResponse response, LoggerProvider loggerProvider, String message) {
```

Comment:
Can you add comments/ javadocs above that explain the difference between `logResponse` vs `logResponsePayload` that that generally both are invoked one after another.

something like: 
LogResponse -> Status Code, Message from Server, Headers
LogResponesPayload -> The parsed json response

Reply:
[e1d9a78](https://github.com/googleapis/google-auth-library-java/pull/1586/commits/e1d9a78d76f037690e87a7fbbe8314f0b4f3cbf8)